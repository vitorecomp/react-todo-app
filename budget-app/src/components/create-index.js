import React from 'react'


import { connect } from 'react-redux'
import { addExpense } from '../actions/expenses.actions'

import ExpenseForm from './expense-form'


const CreatePage = (props) => (
	<div>
		<h1>Create Page</h1>
		<ExpenseForm
			onSubmit={(expense) => {
				props.dispatch(addExpense(expense))
				props.history.push('/')
			}}
		/>
	</div>

)

export default connect()(CreatePage);