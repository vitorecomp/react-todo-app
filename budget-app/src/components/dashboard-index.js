import React from 'react'

import ExpenseList from './expense-list'
import ExpenseFilter from './expense-list-filters'

const DashPage = () => (
	<div>
		<h1>Dash </h1>
		<div>
			<ExpenseFilter />
			<ExpenseList />
		</div>
	</div>
)

export default DashPage;