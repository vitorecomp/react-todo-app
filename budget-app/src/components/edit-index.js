import React from 'react'

import { connect } from 'react-redux'
import { editExpense, removeExpense } from '../actions/expenses.actions'

import ExpenseForm from './expense-form'

const EditPage = (props) => (
	<div>
		<ExpenseForm
			expense={props.expense}
			onSubmit={(expense) => {
				props.dispatch(editExpense(props.match.params.id, expense))
				props.history.push('/')
			}}
		/>
		<button onClick={(e) => {
			props.dispatch(removeExpense(props.match.params.id))
			props.history.push('/')
		}}>
			Remove
		</button>
	</div>
)

const mapStateToProps = (state, props) => {
	return {
		expense: state.expenses.find((expense) => expense.id === props.match.params.id)
	}
}

export default connect(mapStateToProps)(EditPage)