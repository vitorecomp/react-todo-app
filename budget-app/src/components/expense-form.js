import React from 'react'

import moment from 'moment'
import { SingleDatePicker } from 'react-dates'
import 'react-dates/lib/css/_datepicker.css'
import 'react-dates/initialize'

const now = moment();
console.log(now.format('MMM Do, YYYY'))


export default class ExpenseForm extends React.Component {


	constructor(props) {
		super(props)
		this.state = {
			description: props.expense ? props.expense.description : '',
			note: props.expense ? props.expense.note : '',
			amount: props.expense ? (props.expense.amount / 100).toString() : '',
			createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
			calendarFocus: false,
			error: null
		}
	}

	onDescriptionChange = (e) => {
		const description = e.target.value
		this.setState(() => ({
			description
		}))
	}

	onNoteChange = (e) => {
		const note = e.target.value
		this.setState(() => ({
			note
		}))
	}

	onAmountChange = (e) => {
		const amount = e.target.value
		if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
			this.setState(() => ({
				amount
			}))
		}

	}

	onDateChange = (createdAt) => {
		if (!createdAt) {
			return
		}
		this.setState({
			createdAt
		})
	}

	onFocusChange = ({ focused }) => {
		this.setState({
			calendarFocus: focused
		})
	}

	onSubmit = (e) => {
		e.preventDefault();

		if (!this.state.description || !this.state.amount) {
			return this.setState(() => ({ error: 'Please provide a description and amount.' }))
		}

		this.setState(() => ({ error: null }))
		this.props.onSubmit({
			description: this.state.description,
			amount: parseFloat(this.state.amount, 10),
			createdAt: this.state.createdAt.valueOf(),
			note: this.state.note
		});
	}

	render() {
		return (
			<div>

				{this.state.error && <p>{this.state.error}</p>}

				<form onSubmit={this.onSubmit}>
					<input
						type="text"
						placeholder="Description"
						autoFocus
						value={this.state.description}
						onChange={this.onDescriptionChange}
					/>

					<input
						type="text"
						placeholder="Amount"
						value={this.state.amount}
						onChange={this.onAmountChange}
					/>

					<SingleDatePicker
						numberOfMonths={1}
						isOutsideRange={() => false}
						date={this.state.createdAt}
						onDateChange={this.onDateChange}
						focused={this.state.calendarFocus}
						onFocusChange={this.onFocusChange}
					/>

					<textarea
						placeholder="Add a note for your expense(Optional)"
						value={this.state.note}
						onChange={this.onNoteChange}
					/>
					<button>Submit</button>
				</form>
			</div>
		)
	}
}