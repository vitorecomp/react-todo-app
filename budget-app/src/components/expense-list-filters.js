import React from 'react'
import { connect } from 'react-redux'

import { DateRangePicker } from 'react-dates'
import 'react-dates/lib/css/_datepicker.css'
import 'react-dates/initialize'

import {
	setTextFilter,
	sortByAmount,
	sortByDate,
	setStartDate,
	setEndDate
} from '../actions/filters.action'




class ExpenseListFilter extends React.Component {
	state = {
		calendarFocus: null,
	}

	onDatesChange = ({ startDate, endDate }) => {
		this.props.dispatch(setStartDate(startDate))
		this.props.dispatch(setEndDate(endDate))
	}

	onFocusChange = (calendarFocus) => {
		this.setState(() => ({ calendarFocus }))
	}
	render() {
		return (<div>
			<input type="text" value={this.props.filters.text} onChange={(e) => {
				props.dispatch(setTextFilter(e.target.value));
			}} />

			<select onChange={(e) => {
				switch (e.target.value) {
					case 'date':
						this.props.dispatch(sortByDate())
					case 'amount':
						this.props.dispatch(sortByAmount())
				}
			}}>
				<option value='date'>Date</option>
				<option value='amount'>Amount</option>
			</select>
			<DateRangePicker
				isOutsideRange={() => false}
				numberOfMonths={1}

				startDate={this.props.filters.startDate}
				startDateId="start-filter-date"
				endDate={this.props.filters.endDate}
				endDateId="end-filter-date"

				onDatesChange={this.onDatesChange}

				focusedInput={this.state.calendarFocus}
				onFocusChange={this.onFocusChange}

				showClearDates={true}
			/>
		</div>)
	}
}

const mapStateToProps = (state) => {
	return {
		filters: state.filters
	}
}

export default connect(mapStateToProps)(ExpenseListFilter)