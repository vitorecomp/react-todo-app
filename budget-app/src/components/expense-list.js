import React from 'react'
import { connect } from 'react-redux'

import Expense from './expense-item'

import { getVisibleExpenses } from '../selectors/expenses.selectors'

export const ExpenseList = ({ expenses }) => {
	return <div>
		{expenses.length > 0 ?
			expenses.map((e, id) => <Expense key={id} {...e} />) :
			<p>No expenses, please add one!</p>}
	</div>
}
const mapStateProps = (state) => {
	return {
		expenses: getVisibleExpenses(state.expenses, state.filters)
	}
}
export default connect(mapStateProps)(ExpenseList);
