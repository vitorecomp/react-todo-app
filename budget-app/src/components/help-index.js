import React from 'react'

const HelpPage = () => (
	<div> Help Dash </div>
)

export default HelpPage;