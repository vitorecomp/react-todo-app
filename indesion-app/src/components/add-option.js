import React from 'react'

export default class AddOption extends React.Component {
	state = {
		error: null
	}

	onFormSumit = (e) => {
		e.preventDefault()
		const option = e.target.elements.option.value.trim()
		let error = this.props.handleAddOption(option)
		this.setState(() => ({ error }))
		if (!error) {
			e.target.elements.option.value = ''
		}
	}

	render() {
		return <div>
			{this.state.error ? <p className="add-option-error"> {this.state.error}</p> : null}
			<form
				className="add-option"
				onSubmit={this.onFormSumit}>
				<input className="add-option__input" type="text" name="option"></input>
				<button className='button'>
					Add the Option
			</button>
			</form>
		</div >
	}
}
