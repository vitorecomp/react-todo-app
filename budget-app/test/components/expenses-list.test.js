import React from 'react'
import { shallow } from 'enzyme'

import { ExpenseList } from '../../src/components/expense-list'
import expenses from '../fixtures/expense.fixtures'


test('should render Expense list correctly, with expenses', () => {
	const wrapper = shallow(<ExpenseList expenses={expenses} />)
	expect(wrapper).toMatchSnapshot();
})

test('should render Expense list correctly, with empty array', () => {
	const wrapper = shallow(<ExpenseList expenses={[]} />)
	expect(wrapper).toMatchSnapshot();
})
