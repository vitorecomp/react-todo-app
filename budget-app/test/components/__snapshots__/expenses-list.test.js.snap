// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Expense list correctly, with empty array 1`] = `
<div>
  <p>
    No expenses, please add one!
  </p>
</div>
`;

exports[`should render Expense list correctly, with expenses 1`] = `
<div>
  <Expense
    amount={195}
    createdAt={0}
    description="Gun"
    id={1}
    key="0"
    note=""
  />
  <Expense
    amount={196}
    createdAt={-345600000}
    description="Rent"
    id={2}
    key="1"
    note=""
  />
  <Expense
    amount={190}
    createdAt={345600000}
    description="Credit"
    id={3}
    key="2"
    note=""
  />
</div>
`;
